src/SupabaseClient.tsx

```
@@ -0,0 +1,51 @@
+import { createClient } from "@supabase/supabase-js";
+import { Measurement } from "./interfaceMeasurement";
+
+export const supabase = createClient(
+  "https://burmmhznjahkpzyzltgg.supabase.co",
+  process.env.REACT_APP_SUPABASE_API_KEY ?? ""
+);
+
+export async function addMeasurements(measurement: Measurement) {
+  let { data, error } = await supabase
+    .from("reading")
+    .insert([measurement])
+    .select("*");
+
+  if (error != null) {
+    throw new Error(error.message);
+  }
+
+  if (data === null) {
+    throw new Error("Error, data cant be null");
```

After much thought i don't think that's a good approach as it is an async function, I don't know how that behaves with `throw` as it might swallow the exception and the user might not see anything from it. Would rather just to return the Promise to the caller of `addMeasurements` and handle excecptions there with something alone these lines:

```js
addMeasurement(measurement)
  .then(() => reloadOrWhatever())
  .catch((error) => setError(error)); // setError would be a function returned from a useState call and display the error for some seconds to the user.
```